### PR TITLE
AMD SOF based generic SoundWire machine driver

### PR DESCRIPTION
**Pull request by vijendarmukunda**
This patch series includes below listed changes.
1) Move and rename Common helper functions in Intel machine driver to common place holder which can be made generic.
2) Move and rename Codec helper functions which can be made generic.
3) Move the common Codec info list which is generic
4) Add pci revision id field in mach_params for platform distinguish 
5) AMD SOF based SoundWire machine generic driver.

This series is rebase version of https://github.com/thesofproject/linux/pull/4955 .

### CODE REVIEW

File: sound/soc/sdw_utils/soc_sdw_utils.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+int asoc_sdw_startup(struct snd_pcm_substream *substream);
+int asoc_sdw_prepare(struct snd_pcm_substream *substream);
+int asoc_sdw_prepare(struct snd_pcm_substream *substream);
+int asoc_sdw_trigger(struct snd_pcm_substream *substream, int cmd);
+int asoc_sdw_hw_params(struct snd_pcm_substream *substream,
+		       struct snd_pcm_hw_params *params);
+int asoc_sdw_hw_free(struct snd_pcm_substream *substream);
+void asoc_sdw_shutdown(struct snd_pcm_substream *substream);
+
+#endif
```
Comment (plbossart) [inline sound/soc/sdw_utils/soc_sdw_utils.h:207]: maybe move to include/sound/ ?

File: sound/soc/sdw_utils/soc_sdw_utils.h [hunk 2]
```diff
@@ -10,6 +10,50 @@
...
+		     struct snd_soc_dai_link *dai_links,
+		     struct sof_sdw_codec_info *info,
+		     bool playback);
+	int (*exit)(struct snd_soc_card *card, struct snd_soc_dai_link *dai_link);
+	int (*rtd_init)(struct snd_soc_pcm_runtime *rtd, struct snd_soc_dai *dai);
+	bool rtd_init_done; /* Indicate that the rtd_init callback is done */
+	unsigned long quirk;
+};
+
+struct sof_sdw_codec_info {
```
Comment (plbossart) [inline sound/soc/sdw_utils/soc_sdw_utils.h]: rename as soc_sdw_codec_info ?

File: sound/soc/sdw_utils/soc_sdw_utils.h [hunk 3]
```diff
@@ -10,6 +10,50 @@
 #define SOC_SDW_UTILS_H
 
 #include <sound/soc.h>
+#include <sound/soc-acpi.h>
+
+#define SOF_SDW_MAX_DAI_NUM             8
+
+struct sof_sdw_codec_info;
+
+struct sof_sdw_dai_info {
```
Comment (plbossart) [inline sound/soc/sdw_utils/soc_sdw_utils.h]: rename as sof_sdw_dai_info?

File: sound/soc/sdw_utils/soc_sdw_utils.h [hunk 4]
```diff
@@ -10,6 +10,50 @@
 #define SOC_SDW_UTILS_H
 
 #include <sound/soc.h>
+#include <sound/soc-acpi.h>
+
+#define SOF_SDW_MAX_DAI_NUM             8
```
Comment (plbossart) [inline sound/soc/sdw_utils/soc_sdw_utils.h:15]: Rename macro 'SOF_SDW_MAX_DAI_NUM' as 'SOC_SDW_MAX_DAI_NUM'.

that seems to be missed?

File: sound/soc/amd/ps/ps-sdw-mach.c [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+	sdw_shutdown_stream(substream);
+}
+
+static const struct snd_soc_ops acp63_sdw_ops = {
+	.startup = acp63_sdw_startup,
+	.prepare = acp63_sdw_prepare,
+	.trigger = acp63_sdw_trigger,
+	.hw_free = acp63_sdw_hw_free,
+	.shutdown = acp63_sdw_shutdown,
+};
```
Comment (plbossart) [inline sound/soc/amd/ps/ps-sdw-mach.c]: why can't you use the functions you added earlier, e.g.

````c
static const struct snd_soc_ops acp63_sdw_ops = {
	.startup = asoc_sdw_startup,
	.prepare = asoc_sdw_prepare,
	.trigger = asoc_sdw_trigger,
	.hw_params = asoc_sdw_hw_params,
	.hw_free = asoc_sdw_hw_free,
	.shutdown = asoc_sdw_shutdown,
};
````

File: sound/soc/amd/ps/ps-sdw-mach.c [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+static int acp63_sdw_probe(struct platform_device *pdev)
+{
+	int ret;
+	struct snd_soc_card *card;
+
+	dev_dbg(&pdev->dev, "entry=%s\n", __func__);
+	card = &acp63_sdw_card;
+	card->dev = &pdev->dev;
+	usleep_range(200 * 1000, 250 * 1000);
```
Comment (plbossart) [inline sound/soc/amd/ps/ps-sdw-mach.c]: oh that seems so suspicious right there....

That's a NAK from me, there shouldn't be any hard-coded sleep time in a probe, that's a clear sign of something fishy.

File: sound/soc/amd/ps/ps-sdw-mach.c [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+	},
+	.probe = acp63_sdw_probe,
+};
+
+module_platform_driver(acp63_sdw_mach_driver);
+
+MODULE_AUTHOR("Vijendar.Mukunda@amd.com");
+MODULE_DESCRIPTION("TAS2783 audio support");
+MODULE_LICENSE("GPL");
+MODULE_ALIAS("platform:" DRV_NAME);
```
Comment (plbossart) [inline sound/soc/amd/ps/ps-sdw-mach.c]: and what does this have to do with "ASoC: intel/sdw_utils: move rtk jack common helper functions" anyways, that's a patch split or git squash that didn't go well.

File: sound/soc/sdw_utils/soc_sdw_utils.h [hunk 8]
```diff
@@ -13,6 +13,8 @@
 #include <sound/soc-acpi.h>
 
 #define SOF_SDW_MAX_DAI_NUM             8
+#define MAX_NO_PROPS 2
```
Comment (plbossart) [inline sound/soc/sdw_utils/soc_sdw_utils.h:16]: SOC_SDW prefix

File: sound/soc/sdw_utils/soc_sdw_utils.h [hunk 9]
```diff
@@ -42,23 +42,23 @@ struct sof_sdw_dai_info {
...
+struct asoc_sdw_codec_info {
 	const int part_id;
 	const int version_id;
 	const char *codec_name;
 	int amp_num;
 	const u8 acpi_id[ACPI_ID_LEN];
 	const bool ignore_internal_dmic;
 	const struct snd_soc_ops *ops;
-	struct sof_sdw_dai_info dais[SOF_SDW_MAX_DAI_NUM];
+	struct asoc_sdw_dai_info dais[SOF_SDW_MAX_DAI_NUM];
```
Comment (plbossart) [inline sound/soc/sdw_utils/soc_sdw_utils.h:72]: this rename should have been done earlier IMHO

Comment (plbossart) [inline sound/soc/sdw_utils/soc_sdw_utils.h:72]: and you still have SOF_SDW in the define for the size.

File: sound/soc/sdw_utils/soc_sdw_utils.h [hunk 10]
```diff
@@ -113,6 +116,15 @@ const char *asoc_sdw_get_codec_name(struct device *dev,
...
 
+struct asoc_sdw_codec_info *asoc_sdw_find_codec_info_part(const u64 adr);
+
+struct asoc_sdw_codec_info *asoc_sdw_find_codec_info_acpi(const u8 *acpi_id);
+
+struct asoc_sdw_codec_info *asoc_sdw_find_codec_info_dai(const char *dai_name,
+							 int *dai_index);
+
+int asoc_sdw_rtd_init(struct snd_soc_pcm_runtime *rtd);
+
```
Comment (plbossart) [inline sound/soc/sdw_utils/soc_sdw_utils.h:127]: Now that I think about it, it's probably easier to do the renames in one shot

a) rename everything in sof_sdw. It'll be massively different but that's just a rename.
b) move code in the utils stuff.
c) add new things.

That will reduce the scope for errors/omissions.

**Review by plbossart — CHANGES_REQUESTED** [review]
I am having second thoughts on the incremental renames, which contains omissions.
It might be just easier to rename everything in sof_sdw, then do the code moves.

Also the PS stuff is badly broken, a sleep as the first step in a probe isn't going to fly.

File: sound/soc/amd/ps/ps-sdw-mach.c (hunk 5 above)
Comment (vijendarmukunda) [reply sound/soc/amd/ps/ps-sdw-mach.c]: It's my bad. It shouldn't be included in this patch.  This is a just test version machine driver file for PS platform. 
We already working on implementing legacy machine driver similar to sof based soundwire generic machine driver.
Will drop these changes

File: sound/soc/sdw_utils/soc_sdw_utils.h (hunk 3 above)
Comment (vijendarmukunda) [reply sound/soc/sdw_utils/soc_sdw_utils.h]: We have renamed in it later patch.

File: sound/soc/sdw_utils/soc_sdw_utils.h (hunk 2 above)
Comment (vijendarmukunda) [reply sound/soc/sdw_utils/soc_sdw_utils.h]: already renamed in later patch.

File: sound/soc/amd/ps/ps-sdw-mach.c (hunk 7 above)
Comment (vijendarmukunda) [reply sound/soc/amd/ps/ps-sdw-mach.c]: Please ignore this file changes. Will drop this file.

File: sound/soc/sdw_utils/soc_sdw_utils.h (hunk 8 above)
Comment (vijendarmukunda) [reply sound/soc/sdw_utils/soc_sdw_utils.h:16]: will fix it.

File: sound/soc/sdw_utils/soc_sdw_utils.h (hunk 9 above)
Comment (vijendarmukunda) [reply sound/soc/sdw_utils/soc_sdw_utils.h:72]: In later patch we redefined the macro.

File: sound/soc/sdw_utils/soc_sdw_utils.h (hunk 10 above)
Comment (vijendarmukunda) [reply sound/soc/sdw_utils/soc_sdw_utils.h:127]: Agreed. will make the changes.

File: sound/soc/amd/ps/ps-sdw-mach.c (hunk 6 above)
Comment (vijendarmukunda) [reply sound/soc/amd/ps/ps-sdw-mach.c]: Sorry this is a test version file added by mistake. We are working on implementing Soundwire generic machine driver for legacy(NO DSP) stack in similar to SOF based Soundwire generic machine driver. This file will be dropped.

**Comment by vijendarmukunda** [comment]
Removed ps-sdw-mach.c file and rebased the patch series and posted V2 series.
Will post the V3 patch series after fixing the review comments and also will resolve the merge conflict in sof_sdw.c file.

**Comment by vijendarmukunda** [comment]
Requires patches rebase to resolve the merge conflict and needs to redefine the patch series.will open the new PR with fresh patch series

**Pull request closed without merging**